internal/pkg/statemanager/state.go

```diff
@@ -14,19 +16,30 @@ import (
 // We call what the plugin created a ResourceStatus, and which is stored as part of the state.
 type ResourceStatus map[string]interface{}
 
+type ResourceOutputs map[string]interface{}
+
 // State is the single component's state.
 type State struct {
-	Name           string                 `yaml:"name"`
-	InstanceID     string                 `yaml:"instanceID"`
-	DependsOn      []string               `yaml:"dependsOn"`
-	Options        map[string]interface{} `yaml:"options"`
-	ResourceStatus ResourceStatus         `yaml:"resourceStatus"`
+	Name           string                   `yaml:"name"`
+	InstanceID     string                   `yaml:"instanceID"`
+	DependsOn      []string                 `yaml:"dependsOn"`
+	Options        configmanager.RawOptions `yaml:"options"`
+	ResourceStatus ResourceStatus           `yaml:"resourceStatus"`
 }
 
-func (rs *ResourceStatus) SetOutputs(outputs map[string]interface{}) {
+func (rs *ResourceStatus) SetOutputs(outputs ResourceOutputs) {
```

Comment:
since `type ResourceStatus map[string]interface{}` and map is originally a pointer, it is unnecessary to use `func (rs *ResourceStatus)`